visualization/app/codeCharta/services/loadInitialFile/loadInitialFile.service.ts

```diff
@@ -95,21 +95,44 @@ export class LoadInitialFileService {
 			const urlNameDataPairCheckSums = urlNameDataPairs.map(urlNameDataPair => urlNameDataPair.content.fileChecksum)
 			const savedNameDataPairCheckSums = savedNameDataPairs.map(savedNameDataPair => savedNameDataPair.content.fileChecksum)
 			if (stringify(urlNameDataPairCheckSums) === stringify(savedNameDataPairCheckSums)) {
-				this.applyAppSettings(savedCcState.appSettings)
-				this.loadFileService.loadFiles(savedNameDataPairs)
-				this.store.dispatch(setFiles({ value: savedFileStates }))
-				this.applyAllSettings(savedCcState)
-				this.setRenderStateFromUrl()
+				this.applySettingsAndFilesFromSavedState(savedCcState, savedNameDataPairs)
 			} else {
-				this.applyAllSettings(savedCcState)
-				this.loadFileService.loadFiles(urlNameDataPairs)
-				this.setRenderStateFromUrl()
+				this.applySettingsFromSavedState(savedCcState, urlNameDataPairs)
 			}
+			this.setRenderStateFromUrl()
 		} catch (error) {
 			await this.handleErrorLoadFilesFromQueryParams(error as Error)
 		}
 	}
 
+	private applySettingsAndFilesFromSavedState(savedCcState: CcState, savedNameDataPairs: NameDataPair[]) {
+		const missingPropertiesInSavedCcState = []
+
+		const missingAppSettings = this.applyAppSettings(savedCcState.appSettings)
+		missingPropertiesInSavedCcState.push(...missingAppSettings)
+
+		this.loadFileService.loadFiles(savedNameDataPairs)
+		const missingFileSettings = this.applyFileSettings(savedCcState.fileSettings)
+		missingPropertiesInSavedCcState.push(...missingFileSettings)
+		const missingDynamicSettings = this.applyDynamicSettings(savedCcState.dynamicSettings)
+		missingPropertiesInSavedCcState.push(...missingDynamicSettings)
+
+		if (missingPropertiesInSavedCcState.length > 0) {
+			this.showErrorDialogForMissingProperties(missingPropertiesInSavedCcState)
+		}
+	}
+
+	private applySettingsFromSavedState(savedCcState: CcState, urlNameDataPairs: NameDataPair[]) {
+		this.applyAllSettings(savedCcState)
+		this.loadFileService.loadFiles(urlNameDataPairs)
+	}
+
+	private showErrorDialogForMissingProperties(missingPropertiesInSavedCcState) {
+		const title = "All properties expect the following were restored successfully."
```

Comment:
Is this shown to the userse? I would rephrase it a little bit, please.
I would give a understandable summary like "The previous state could not be restored after reloading the page."
Then I would give more details about it in a separate paragraph or collpased list with "see more details".